backend/src/main.py

```
@@ -81,12 +72,87 @@ def internal_error(error):
+        "data_fetcher": data_fetcher_service,
+        "ai_analysis": ai_analysis_service,
+        "auto_trader": auto_trader_service
+    })
+
+    # Start background asyncio services in a dedicated thread
+    bg_thread = threading.Thread(target=start_async_loop, daemon=True)
+    bg_thread.start()
+
+    # Run the Flask-SocketIO app
+    logger.info("Starting Flask-SocketIO server on port 5000...")
+    socketio.run(app, host='0.0.0.0', port=5000)
```

Ruff (0.15.12)

[error] Possible binding to all interfaces

(S104)